src/Servant/Multipart.hs

```diff
@@ -246,10 +258,10 @@ class FromMultipart tag a where
   --   in a list of textual inputs and another list for
   --   files, try to extract a value of type @a@. When
   --   extraction fails, servant errors out with status code 400.
-  fromMultipart :: MultipartData tag -> Maybe a
+  fromMultipart :: MultipartData tag -> Either String a
```

Comment:
There are ways to make this change easier for the library users, but I'm not sure it should be.  Perhaps it's good to force everybody to take a look at this if they can't somehow produce a useful error here; if not, it's still straight-forward enough to do something like `maybe (Left "no error info, sorry") Right` to the maybe-parser.

Yes, I think it should be a breaking change.  @arianvp, @dmjio anybody: any opinions?